SSO Doc (#869)
huggingface/hub-docs: docs/hub/security-sso.md

# Single Sign-On (SSO)

> The Hugging Face Hub gives you the ability to implement mandatory Single Sign-On (SSO) for members of your organization. You can enable your SSO provider for a specific email domain (for example, @example.com). When enabled, users with an email address that matches the domain must log in to huggingface.co using your SSO provider to access your organization.

Reviewer: ```suggestion
The Hugging Face Hub gives you the ability to implement mandatory Single Sign-On (SSO) for your organization.
```
(the rest is a bit redundant with the following section I think)

<Tip warning={true}>

> This feature is available for Enterprise Hub. <a href="https://huggingface.co/enterprise" target="_blank">For more information, please contact our sales team</a>.

Reviewer: ```suggestion
This feature is part of the <a href="https://huggingface.co/enterprise" target="_blank">Enterprise Hub</a>.
```

<div class="flex justify-center">
<img class="block dark:hidden" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/security-sso-prompt.png"/>
<img class="hidden dark:block" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/security-sso-prompt-dark.png"/>
</div>

- Org members logged in on hf.co must authenticate through SSO even to access public content from your org. Such content will still be accessible to anyone outside the org. SSO only applies to your org, meaning your org members can still belong to other orgs on hf.co.
- Role mapping is supported and determined by your IdP and your settings. Enabling role mapping means you can't assign roles manually in the org.

## Configuring SSO

To successfully enable SSO on the Hugging Face Hub for your organization, you need to set up an OIDC application in your IdP. We will provide a callback URL pointing to the `huggingface.co` domain. Once you create the application, you will need to provide us with the following information:

- Client ID
- Client Secret
- Issuer URL

## Supported Identity Providers

> You can easily integrate Hugging Face Hub with a variety of Identity Providers, such as Okta, OneLogin or Azure Active Directory (Azure AD). Hugging Face Hub can work with any OIDC-compliant Identity Provider.

Reviewer 2: can we say something about upcoming support for SAML?

# Single sign-on (SSO)

> With Hugging Face Hub, you have the option to implement mandatory Single Sign-On (SSO) for members of your organization. You can activate your SSO provider for a particular email domain (e.g., @example.com). Users with an email address that matches the domain must log in on huggingface.co using your SSO provider to access your organization.

Reviewer 3: ```suggestion
The Hugging Face Hub gives you the ability to implement mandatory Single Sign-On (SSO) for members of your organization. You can enable your SSO provider for a specific email domain (for example, @example.com). When enabled, users with an email address that matches the domain must log in to huggingface.co using your SSO provider to access your organization.
```

<Tip warning={true}>

> This feature is available for Enterprise Hub. <a href="mailto:sales@huggingface.co" target="_blank">For more information, please contact our sales team</a>.

Author: ```suggestion
This feature is available for Enterprise Hub. <a href="https://huggingface.co/enterprise" target="_blank">For more information, please contact our sales team</a>.
```

</Tip>

## Rules

Once SSO is enabled for the plan, the following rules apply to the end users:

- Your members will see a banner prompting them to authenticate through your chosen IdP to access your organization.

<div class="flex justify-center">
<img class="block dark:hidden" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/security-sso-prompt.png"/>
<img class="hidden dark:block" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/security-sso-prompt-dark.png"/>
</div>

> - If an end-user is a member of any organization outside yours, they are still required to log in via SSO as soon as the feature is enabled to access public content of your organization. However, they will be able to access other teams.

Reviewer 3: ```suggestion
- If an end user is a member of another organization, they will still need to log in via SSO as soon as the feature is enabled to access your organization's public content. However, they will be able to access other teams.
```

> - If you enable the role mapping feature, the admin from your organization won't be able to change their role. It will be determined by your IdP and your settings.

Author: it's done manually as long as we don't have self-service yet

## Configuring SSO

To successfully enable Single Sign-On (SSO) on the Hugging Face Hub for your organization, we will need to set up an OIDC application in your IdP. We will provide you with a callback URL pointing to the `huggingface.co` domain. After creating the application, we will need this information to make your SSO implementation work:

- Client ID
- Client Secret
- Issuer URL

## Supported SSO Partners

> You can easily integrate Hugging Face Hub with a variety of SSO providers, and we have listed a few examples below:
> 
> 1. Okta
> 2. OneLogin
> 3. Azure Active Directory (Azure AD)
> 
> These are just some of the popular examples, but Hugging Face Hub can work with any OIDC-compliant SSO system.

Reviewer: ```suggestion

You can easily integrate Hugging Face Hub with a variety of Identity Providers, such as Okta, OneLogin or Azure Active Directory (Azure AD). Hugging Face Hub can work with any OIDC-compliant Identity Provider.
```
